wled00/FX_2Dfcn.cpp

```diff
@@ -458,43 +414,69 @@ void Segment::box_blur(unsigned radius, bool smear) {
   delete[] tmpBSum;
   delete[] tmpWSum;
 }
-
+*/
 void Segment::moveX(int delta, bool wrap) {
-  if (!isActive()) return; // not active
+  if (!isActive() || !delta) return; // not active
   const int vW = vWidth();   // segment width in logical pixels (can be 0 if segment is inactive)
   const int vH = vHeight();  // segment height in logical pixels (is always >= 1)
-  if (!delta || abs(delta) >= vW) return;
+  int absDelta = abs(delta);
+  if (absDelta >= vW) return;
   uint32_t newPxCol[vW];
+  int newDelta;
```

Comment:
I do not see the point in modifying `move()` functions, especially with convoluted math that is hard to follow.
How much speed does it gain?